Index: 04.for/main.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package main\n\nimport (\n\t\"fmt\"\n)\n\nfunc main() {\n\tfmt.Println(\"Round 1\")\n\tfor i := 0; i < 5; i++ {\n\t\tfmt.Println(\"你好，Golang!\")\n\t}\n\n\tfmt.Println(\"Round 2\")\n\tj := 0\n\tfor ; j < 5; j++ {\n\t\tfmt.Println(\"Hello, Golang!\")\n\t}\n\n\tfmt.Println(\"Round 3\")\n\tk := 0\n\tfor k < 5 { // 省略;\n\t\tfmt.Println(\"Hi, Golang!\")\n\t\tk++\n\t}\n\n\tfmt.Println(\"Round 4\")\n\tm := 0\n\tfor {\n\t\tfmt.Println(\"Yo, Golang!\")\n\t\tm++\n\t\tif m >= 5 {\n\t\t\tbreak // 打断跳出for循环\n\t\t}\n\t}\n\n\tfmt.Println(\"Round 5\")\n\tm = 0\n\tfor {\n\t\tfmt.Println(\"Hello, Golang!\", m)\n\t\tm++\n\t\tif m >= 10 {\n\t\t\tbreak\n\t\t}\n\t\tif m%2 == 0 {\n\t\t\tfmt.Println(\"Round 5，被continue跳过\")\n\t\t\tcontinue\n\t\t}\n\t\tfmt.Println(\"Round 5，练习跳过\", m)\n\t}\n\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/04.for/main.go b/04.for/main.go
--- a/04.for/main.go	(revision c392e9c8836f7e168ff2fce4c5aa7a67dc0f3e6a)
+++ b/04.for/main.go	(date 1638875548849)
@@ -34,18 +34,18 @@
 	}
 
 	fmt.Println("Round 5")
-	m = 0
+	n := 0
 	for {
-		fmt.Println("Hello, Golang!", m)
-		m++
-		if m >= 10 {
+		fmt.Println("Hello, Golang!", n)
+		n++
+		if n >= 10 { // n=10时结束循环
 			break
 		}
-		if m%2 == 0 {
+		if n%2 == 0 {
 			fmt.Println("Round 5，被continue跳过")
-			continue
+			continue // n为偶数时，跳过当前循环的剩余语句，继续进行下一轮循环
 		}
-		fmt.Println("Round 5，练习跳过", m)
+		fmt.Println("Round 5，练习跳过", n)
 	}
 
 }
